tests/terraform/e2e_controller/main.tf

```diff
@@ -0,0 +1,78 @@
+data "aws_ami" "latest_controller" {
+  most_recent = true
+  owners      = ["600530653622"]
+
+  filter {
+    name   = "name"
+    values = ["avx-controller-g4-*"]
+  }
+}
+
+# try to satisfy the controller's password strength meter by combining
+# two passwords with a special char
+resource "random_password" "admin" {
+  count = 3
+
+  length      = 6
+  special     = false
+  min_numeric = 1
+}
+
+locals {
+  admin_password = nonsensitive(join("-", random_password.admin[*].result))
+}
+
+module "my_controller" {
+  source = "git::https://github.com/AviatrixSystems/terraform-aviatrix-aws-controlplane?ref=no-cft"
+
+  account_email             = var.admin_email
+  customer_id               = var.customer_id
+  controller_admin_email    = var.admin_email
+  controller_admin_password = local.admin_password
+  incoming_ssl_cidrs        = var.incoming_ssl_cidrs
+
+  controller_ami_id = var.controller_ami_id == "" ? data.aws_ami.latest_controller.id : var.controller_ami_id
+  controller_user_data = templatefile("${path.module}/cloud-config.yaml", {
+    software_version          = var.controller_version
+    use_containerized_gateway = var.use_containerized_gateway ? "true" : "false"
+  })
+  controller_wait_for_setup_duration = "0s"
```

Comment:
Curious what this does and why we set it to 0s ?  I couldn't find in the referenced module.

Reply:
the controller-init module is common for pre-containerized and containerized controllers. In the pre-containerized case, the "initial_setup" API finishes after phase 1 (download software) but we still need some time for phase 2 to complete, so it sleeps for 10 minutes (!) before continuing. In the containerized case we don't need this 10 minute sleep.

Reply:
Nice